hls4ml/model/layers.py

```diff
@@ -880,11 +880,17 @@ def initialize(self):
             self.add_output_variable(state_shape, state_dims, out_name=self.outputs[1], var_name='layer{index}_h', type_name='layer{index}_h_t')
             self.add_output_variable(state_shape, state_dims, out_name=self.outputs[2], var_name='layer{index}_c', type_name='layer{index}_c_t')
 
-        self.add_weights()
-        self.add_bias()
 
-        recurrent_weight = self.model.get_weights_data(self.name, 'recurrent_kernel')
-        self.add_weights_variable(name='recurrent_weight', var_name='wr{index}', data=recurrent_weight)
+        self.add_weights(quantizer=self.get_attr('weight_quantizer'), compression=self.model.config.get_compression(self))
+        self.add_bias(quantizer=self.get_attr('bias_quantizer'))
+
+        data  = self.model.get_weights_data(self.name, 'kernel')
+        data2 = self.model.get_weights_data(self.name, 'recurrent_kernel')
+        data3 = self.model.get_weights_data(self.name, 'bias')
+
+        self.add_weights_variable(name='weight', var_name='kernel_{index}' , data=data[0][0:self.get_attr('n_out')], quantizer=self.get_attr('weight_quantizer'), compression=None)
+        self.add_weights_variable(name='recurrent_weight', var_name='recurrent_kernel_{index}', data=data2[0:self.get_attr('n_out'),0:self.get_attr('n_out')], quantizer=self.get_attr('weight_quantizer'), compression=None)
+        self.add_weights_variable(name='bias', var_name='bias_{index}', data=data3[0:self.get_attr('n_out')], quantizer=self.get_attr('weight_quantizer'), compression=None)
```

Comment:
Possible duplication? Lines 884-885 already add the 'weight' and 'bias' variables, so 887 & 889 would simply be re-doing the same operation? However, 888 is still needed, as the recurrent weights are not included yet.

Reply:
Correct, removed lines 884 and 885